### DISCUSSION TITLE
向量维度的大小对相似度值的影响

### DISCUSSION THREAD

**Discussion by mmahao** [milvus-io/milvus#35239] [Q&A and General discussion]
Metric Type：IP；
Nprobe：4；
Round Decimals：-1；

之前维度是：384，检索的时候，相似度最高的值就是1；
现在维度是：1536，检索的时候，相似度有时候能到好几千；

目前的业务逻辑是会根据相似度进行数据过滤（之前是比如说用 0.9 来过滤），但是切换到 1536 维度后，这个相似度的值很难确定到一个阈值范围；

所以
维度是会影响检索出来的相似度吗？
有什么方式可以来确定一个阈值范围？
网上搜了一下，有个 归一化 的概念，不知道结合 milvus 适不适用？

对这块比较熟悉的大佬，希望可以帮忙解答一下，感谢🙏

**Comment by yhmo** [comment]
下次提问请尽量使用英文。

向量检索出来的值只能用来找出哪个是最可能是要找的目标，但这些值不适合用来衡量相似的程度。
假设有100个目标，我们用两个不同的模型产生两组100条向量。你用其中一个目标去检索，得到两组topk的相似结果。top1的那条值都是1，但是，top2的值有可能大相径庭。其实不是维度影响。是所用的模型决定了维度，也决定了相似度的这些值的分布。
对于你的问题来说，每个模型，会有不同的阀值，换模型就换阀值。

归一化只是为了让向量能以余弦相似运算来做检索，没有归一化的向量用余弦相似运算无法得到正确的结果，没有归一化的向量可以用欧式距离运算来确定相似度。建议先了解一下欧式距离和余弦相似度运算的数学意义：https://milvus.io/docs/metric.md?tab=floating#Similarity-Metrics

**Reply by mmahao** [reply]
好的，感谢

**Comment by xiaofan-luan** [comment]
如果希望分数是-1，1之间，需要用cosine metrics。或者向量本身要做normalization处理。不同模型产出的向量，阈值本身就是不同的

**Reply by mmahao** [reply]
感谢
